server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -262,6 +279,48 @@ protected void doClose() {
         }
     }
 
+    // Inspects all cluster state elements that contain a hint about what the current repository generation is and updates
+    // #latestKnownRepoGen if a newer than currently known generation is found
+    @Override
+    public void applyClusterState(ClusterChangedEvent event) {
```

Comment:
I thought about that initially but then dropped the idea because that would've meant leaking the specifics of `BlobStoreRepository` into `RepositoriesService`.
But come to think of it now ... doing this would also save a bit of tricky of code in #49060 to get the initialization of the repo on non-master nodes right.
Let's do it -> will do :)

Reply:
> I wonder if `RepositoriesService` should update the relevant repository about changes to their snapshots.

I agree, that seems to be the right thing to do. I'm also not super happy to have BlobStoreRepository have its own lifecycle.

Reply:
Could I warm your heart for cfb104f maybe? :)
I think in the case of the repositories it's better to pass the whole `ClusterState` to the repo instead of figuring out what parts go to what repo in `RepositoriesService`. Otherwise we needlessly waste cycles on the repos that don't require the cluster state (e.g. read-only repos, ccr, ...). Also, in #49060 the parts of the CS that need to be inspected will change, but we may want to keep the current logic from here as BwC fallback which is much easier if we just pass the full `ClusterState` down.
Also, I realized that I actually had to move the cluster state application method to `Repository` because of `FilterRepository` (and upcoming encrypted repo wrapper etc.) so this didn't leak any blob store repo specifics to `RepositoriesService` after all :)